net/dns/manager.go

```diff
@@ -79,6 +84,23 @@ func NewManager(logf logger.Logf, oscfg OSConfigurator, health *health.Tracker,
 		knobs:    knobs,
 		goos:     goos,
 	}
+
+	// Rate limit our attempts to correct our DNS configuration.  The recovery func is triggered
+	// by SERVFAIL errors, which can be occurring in batches at relatively high frequency.
+	limiter := rate.NewLimiter(1.0/5.0, 1)
+
+	// This will recompile the DNS config, which in turn will requery the system
+	// DNS settings.
+	m.resolver.SetServFailRecovery(func() {
```

Comment:
Am I reading this correctly that if a user has a piece of software that happens to regularly poll a bad upstream server, we'll now reconfigure DNS every 5s? Could we limit this to only occur when there are no known upstream DNS servers?

Reply:
It is only called when we try a query and internally return SERVFAIL because we have not nameservers.   Clarified the naming and the comments.